src/NuGet.Clients/VisualStudio.Facade/VisualStudio14.Packages/project.json

```diff
@@ -8,6 +8,9 @@
     "Microsoft.VisualStudio.Services.Client": "14.95.3",
     "Microsoft.VisualStudio.Services.InteractiveClient": "14.95.3",
     "Microsoft.VisualStudio.Shell.14.0": "14.2.25123",
+    "Microsoft.VisualStudio.Shell.Immutable.10.0": "10.0.30319",
+    "Microsoft.VisualStudio.Shell.Immutable.11.0": "11.0.50727",
+    "Microsoft.VisualStudio.Shell.Immutable.12.0": "12.0.21003",
```

Comment:
I believe each one only contains the interfaces new for that version of VS so that the strong name doesn't change.